lib/active_merchant/billing/gateways/garanti.rb

```diff
@@ -251,6 +251,10 @@ def success?(response)
         response[:message] == "Approved"
       end
 
+      def strip_invalid_xml_chars(xml)
+        xml.gsub(/&(?!(?:[a-z]+|#[0-9]+|x[a-zA-Z0-9]+);)/, '&amp;')
```

Comment:
That regex covers basically '&', '<', '>', double quote and apostrophe. Any other html entity (such as &atilde; &ccedil; etc) will have wrongfully converted their '&' with '&amp;'.

www.debuggex.com is a really option to understand complex regex.